models/registro_plan_adquisicion.go

```diff
@@ -152,25 +156,188 @@ func BuscarIndexPorCampo(RegistroPlanAdquisicion []map[string]interface{}, Rubro
+				outputError = map[string]interface{}{
+					"funcion": "IngresoPlanAdquisicion -  movimientosCrud.GetMovimientoProcesoExterno(query, \"\", sortby, order, \"\", limit)",
+					"err":     err,
+					"status":  "502",
+				}
+				return nil, outputError
+			} else {
+				// logs.Debug("Tipo de Resultado: ", reflect.TypeOf(resultado), " - Resultado: ", resultado)
+				var movimientoObtenido map[string]interface{}
+				switch resultado.(type) {
+				case map[string]interface{}:
+					movimientoObtenido = resultado.(map[string]interface{})
+				case []interface{}:
+					movimientoObtenido = resultado.([]interface{})[0].(map[string]interface{})
+				}
+
+				keys := make([]string, 0, len(movimientoObtenido))
+				for k := range movimientoObtenido {
+					keys = append(keys, k)
+				}
+
+				if len(keys) > 0 {
+					// logs.Debug("ID OBTENIDO: ", int(movimientoObtenido["Id"].(float64)))
+					movimientoExternoID = int(movimientoObtenido["Id"].(float64))
+				} else {
+					if movimientoInsertar, err := ObtenerMovimientoProcesoExterno(idPlanAdquisiciones); err != nil {
+						logs.Error(err)
+						return nil, err
+					} else {
+						if idMovimientoInsertado, err := movimientosCrud.CrearMovimientoProcesoExterno(movimientoInsertar); err != nil {
+							logs.Error(err)
+							outputError = errorctrl.Error("IngresoPlanAdquisicion -  movimientosCrud.CrearMovimientoProcesoExterno(movimientoInsertar)", err, "502")
+							return nil, outputError
+						} else {
+							// logs.Debug("ID OBTENIDO: ", int(idMovimientoInsertado.(map[string]interface{})["Body"].(map[string]interface{})["Id"].(float64)))
+							movimientoExternoID = int(idMovimientoInsertado.(map[string]interface{})["Body"].(map[string]interface{})["Id"].(float64))
+						}
+					}
+				}
+
+			}
+
+			if resultado, err := ObtenerRegistroMovimientoInversion(registroPlanAdquisicion, movimientoExternoID); err != nil {
+				logs.Error(err)
+				outputError = map[string]interface{}{
+					"funcion": "IngresoPlanAdquisicion - ObtenerRegistroMovimientoInversion(registroPlanAdquisicion, movimientoExternoID)",
+					"err":     err,
+					"status":  "502",
+				}
+				return nil, outputError
+			} else {
+				if len(resultado) > 0 {
+					if _, err := movimientosCrud.CrearMovimientosDetalle(resultado); err != nil {
+						logs.Error(err)
+						outputError = errorctrl.Error("IngresoPlanAdquisicion -  movimientosCrud.CrearMovimientosDetalle(resultado)", err, "502")
+						return nil, outputError
+					}
+				}
+			}
+			return resultadoPlan, nil
 		}
 	} else {
-		result, errorDatos := IngresoRenglonPlanFuncionamiento(registroPlanAdquisicion)
-		if errorDatos != nil {
-			return nil, errorDatos
+		resultadoPlan, err := IngresoRenglonPlanFuncionamiento(registroPlanAdquisicion)
+		if err != nil {
+			logs.Error(err)
+			outputError := errorctrl.Error("IngresoPlanAdquisicion -  IngresoRenglonPlanFuncionamiento(registroPlanAdquisicion)", err, "502")
+			return nil, outputError
```

Comment:
Si `IngresoRenglonPlanFuncionamiento` retorna un `err`or formateado, se puede dejar así
```suggestion
			return nil, err
```
_a no ser_ que el `err`or retornado deba procesarse de otra forma